src/components/PreviewPage.svelte

```diff
@@ -38,10 +38,10 @@
       // TODO: Cancel pending renders? Accept in-order renders?
       if (hash === updateHash && scale === updateScale) {
         const ctx = canvas.getContext("2d");
-        canvas.width = width;
-        canvas.height = height;
+        canvas.width = width * window.devicePixelRatio;
```

Comment:
`width` and `height` values should be floored.

Reply:
I believe that I have fixed this. I also felt it was more readable to have them in separate variables after adding the floor calls. Please let me know if this isn't the style you'd prefer :))